app/services/time_discrepancy_loader.rb

```diff
@@ -3,23 +3,15 @@
 class TimeDiscrepancyLoader
   DECIMALS_TO_ROUND_TO = 2
 
-  # base_query ==> Expected to be TimerSession.where(user_id: user.id)
-  def initialize(base_query)
-    @base_query = base_query
-  end
-
-  def where_time_not_adding_up
-    build_query
-  end
-
-  private
-
-  # TODO: rewrite to be in SQL rather than ruby
-  def build_query
-    @base_query.reject do |timer_session|
-      hours_to_spend = timer_session.splittable_hours
-      total_hours_recorded = timer_session.time_entries.to_a.sum(&:hours)
-      hours_to_spend.to_d.round(DECIMALS_TO_ROUND_TO) == total_hours_recorded.to_d.round(DECIMALS_TO_ROUND_TO)
-    end
+  def self.uneven_timer_sessions(scope = TimerSession.all)
+    scope.joins(:time_entries)
+         .group(:id)
+         .having(<<~SQL)
+           CAST(SUM(time_entries.hours)
+           AS DECIMAL(10, ${DECIMALS_TO_ROUND_TO}))
+           !=
+           CAST(TIMESTAMPDIFF(SECOND, timer_sessions.timer_start, timer_sessions.timer_end) / 3600.0
+           AS DECIMAL(10, ${DECIMALS_TO_ROUND_TO}))
+         SQL
```

Comment:
Should work both on Postgres and MySQL (not using `EPOCH`)

Reply:
Well it doesn't seem to work when trying to use it:
![image](https://user-images.githubusercontent.com/1435346/235957701-c7944190-707e-487a-ac0d-f4db979768f2.png)

Reply:
What database do you use locally? SQLite?

Reply:
postgres of course you clown :wink:

Reply:
It didn't work on @simon-isler's machine and I tried just the query on heroku where it also failed.